src/stream.jl

```diff
@@ -9,6 +9,13 @@ const CuStream_t = Ptr{Void}
 type CuStream
     handle::CuStream_t
     ctx::CuContext
+    function CuStream(handle::CuStream_t, ctx=CuCurrentContext())
+        obj = new(handle, ctx)
+        block_finalizer(obj, ctx)
+        finalizer(obj, finalize)
+        return obj
+    end
+    CuStream(handle::CuStream_t, ctx::CuContext, _) = new(handle, ctx)
```

Comment:
Why is this necessary? I don't really like having such a three-argument ctor where the third argument is ignored...